requirements.txt

```diff
@@ -1,20 +1,20 @@
-aiofiles>=0.8.0,<=22.1.0
-aiohttp>=3.8.1
-alembic>=1.8.0,<=1.8.1
+aiofiles==22.1.0
+aiohttp==3.8.1
+alembic==1.8.1
 click==8.1.3
-cloudpickle>=2.0.0,<=2.2.0
-dask[distributed]>=2022.6.0
-fastapi>=0.93.0,<=0.94.1
+cloudpickle==2.2.0
+dask[distributed]==2022.9.0
+fastapi==0.94.1
 furl==2.1.3
 networkx==2.8.6
-psutil>=5.9.0,<=5.9.2
-pydantic>=1.10.1,<=1.10.2
+psutil==5.9.2
+pydantic==1.10.2
 python-socketio==5.7.1
-requests>=2.24.0,<=2.28.1
+requests==2.28.1
 simplejson==3.17.6
-sqlalchemy>=1.4.37,<=1.4.41
+sqlalchemy==1.4.41
 sqlalchemy_utils==0.38.3
 toml==0.10.2
-uvicorn[standard]>=0.18.2,<=0.18.3
+uvicorn[standard]==0.18.3
 watchdog==2.2.1
-werkzeug>=2.0.3,<=2.2.2
+werkzeug==2.2.2
```

Comment:
Instead of exactly pinning the versions like this, let's use the [compatible release](https://peps.python.org/pep-0440/#compatible-release) operator so that we're not tied to one version and upgrades are ensured to work due to backwards compatibility.

If it's not too much could you also edit the `tests/requirements.txt` file for the same reason. Much appreciated, thanks 🙏 !

Reply:
In fact, I'm almost leaning towards not using the strict pins at all here and only use the `>=` one. Once/If something breaks due to the dependency upgrade, then we can put it within a range.

Reply:
@arosen93 that's a good point. Although, I'd say that ideally any breakage in dependency should get caught in the nightly run of the test suite that takes place at around 12 am EST 😅 . Plus, we'd also like to know the reason behind that breakage, and whether it is better to improve the code which is using that broken dependency or just restrict the version of that dependency - which in itself might warrant a manual PR.

Reply:
@kessler-frost, ah that makes a lot of sense! I forgot about the nightly test.

So, one final question then. If there's no plan to have version pins in the `requirements.txt`, is there a benefit of having a separate `install_requires` in the `setup.py` after all? Would the best solution to read the `install_requires` from `requirements.txt` like what's in the `develop` branch but to use the less strict versions I've include in the `setup.py` within this PR?

Reply:
Ah, yeah so your issue (and this PR) actually sparked a discussion which led us to consider using a `requirements-ci.txt` instead in the near future. Which, I'd say this `requirements.txt` will either transition into or be a supplement to this. In `install_requires` eventually what we'd like to do is have no version restrictions at all for as many packages as possible. So, until then we should have both `install_requires` and `requirements.txt` which can evolve independently 😄 .